interface/javascript/bw_stage_01.js

```diff
@@ -61,463 +60,446 @@ settings={
+	loader.add('first-image', settings.imagesURL[currentImageID]);
+	loader.add('first-image-neg', settings.imagesNegativeURL[currentImageID]);
+	loader.add('first-filter', settings.dmapsURL[currentDmapID]);
+	loader.load(firstImagesLoadCompleteListener);
 	
-	resizeListener = function(){
-		var image, imageNegativ;
-		
-		app.renderer.resize(window.stageSettings.width, window.stageSettings.height);
-		app.view.style.width=window.stageSettings.width+"px";
-		app.view.style.height=window.stageSettings.height+"px";
-		
-		for(i=0; i<imagesRep.length; i++){
-			image=imagesRep[i];
-			
-			image.x = app.renderer.width / 2;
-			image.y = app.renderer.height / 2;
-			
-			if(window.stageSettings.width/window.stageSettings.height >= settings.imagesBW/settings.imagesBH){
-				image.width = window.stageSettings.width * (1+settings.mouseMoveDecalMaxRatio);
-				image.height = settings.imagesBH * (image.width/settings.imagesBW);
-			}else{
-				image.height = window.stageSettings.height * (1+settings.mouseMoveDecalMaxRatio);
-				image.width = settings.imagesBW * (image.height/settings.imagesBH);
-			}
-			
-			imageNegativ=imagesNegativeRep[i];
-			imageNegativ.x = image.x;
-			imageNegativ.y = image.y;
-			imageNegativ.width = image.width;
-			imageNegativ.height = image.height;
-		}
-		for(i=0; i<settings.dmapsURL.length; i++){
-			dmapsRep[i].sprite.scale.x = dmapsRep[i].sprite.scale.y = settings.dmapsParamaters[i].spriteInitScale;    
-			dmapsRep[i].sprite.x= (dmapsRep[i].sprite.scale.x*settings.filterSize - window.stageSettings.width) * -0.5;
-			dmapsRep[i].sprite.y= (dmapsRep[i].sprite.scale.y*settings.filterSize - window.stageSettings.height) * -0.5;	
-		}  
+	const imagesContainer = new PIXI.Container();
+	imagesColorMatrix = new PIXI.filters.ColorMatrixFilter();
+	imagesContainer.filters = [imagesColorMatrix];
+	
+	for(i=0; i<settings.imagesURL.length; i++){
+		let imageTexture = PIXI.Texture.fromImage(settings.imagesURL[i]);
+		let image = new PIXI.Sprite(imageTexture);
+		image.anchor.set(0.5);
+		imagesContainer.addChild(image);
+		if(i!=currentImageID){ 	image.alpha=0; }
+		imagesRep.push(image);
+		
+		imageTexture = PIXI.Texture.fromImage(settings.imagesNegativeURL[i]);
+		image = new PIXI.Sprite(imageTexture);
+		image.anchor.set(0.5);
+		imagesContainer.addChild(image);
+		image.alpha=0;
+		imagesNegativeRep.push(image);
+	}
+	app.stage.addChild(imagesContainer);
+	
+	currentImage=imagesRep[currentImageID];
+	
+	linesOverlay = new PIXI.Graphics();
+	app.stage.addChild(linesOverlay);
+	drawRandomLines();
+	
+	mouseMoveLineOverlay = new PIXI.Graphics();
+	mouseMoveLineOverlay.props={
+		mouseMoveLineWidthCurrent:0
 	};
+	app.stage.addChild(mouseMoveLineOverlay);
+	animateMouseMoveLineWidth();
 	
-	mouseMoveListener = function ( event ) {
-		mousePos.x=mousePosInStage.x=event.pageX-window.stageSettings.posX;
-		mousePos.y=mousePosInStage.y=event.pageY-window.stageSettings.posY;
+	for(i=0; i<settings.dmapsURL.length; i++){
```

Comment:
should be `for(let i=0; i<settings.dmapsURL.length; i++){` i would not declare increment vars on top of the scope

Reply:
Oh and since its an array why not just use `settings.dmapsURL.forEach()`

Reply:
i keep the for because i need the i to interact with others arrays

Reply:
forEach also give you an `index` ex:

```js
['ta', 'maman', 'en', 'string'].forEach((str, i) => {
   console.log(str, i); // => str: 'ta' i: 0, str: 'maman' i: 1, etc... 
})
```